LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs

```diff
@@ -149,6 +159,47 @@ private async Task<MethodResponse> ClearCacheAsync()
             return new MethodResponse((int)HttpStatusCode.OK);
         }
 
+        private async Task<MethodResponse> DropConnectionAsync(MethodRequest methodRequest)
+        {
+            this.forceClosedConnections.Add(1);
+            ReceivedLoRaCloudToDeviceMessage c2d = null;
+
+            this.logger.LogDebug("Will deserialize '{JsonData}'.", methodRequest.DataAsJson);
+
+            try
+            {
+                c2d = JsonSerializer.Deserialize<ReceivedLoRaCloudToDeviceMessage>(methodRequest.DataAsJson, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+            }
+            catch (Exception ex) when (ex is ArgumentNullException or JsonException)
+            {
+                this.logger.LogError(ex, "Unable to parse Json for direct method '{MethodName}' for device '{DevEui}', message id '{MessageId}'", methodRequest.Name, c2d?.DevEUI, c2d?.MessageId);
+                return new MethodResponse((int)HttpStatusCode.BadRequest);
+            }
+
+            using var cts = methodRequest.ResponseTimeout.HasValue ? new CancellationTokenSource(methodRequest.ResponseTimeout.Value) : null;
```

Comment:
no need to create the cts, if the validation fails on the next line. Hence, create it after you validate it has a valid DevEUI

Reply:
Done in d489a749d8a517f62eb764a7522091ec36527b1c